source/jormungandr/jormungandr/realtime_schedule/sytral.py

```diff
@@ -0,0 +1,181 @@
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error(
+                'systralRT RT service timeout, using base ' 'schedule (error: {}'.format(t),
+                extra={'rt_system_id': unicode(self.rt_system_id)},
+            )
+            raise RealtimeProxyError('timeout')
+        except Exception as e:
+            logging.getLogger(__name__).exception(
+                'systralRT RT error, using base schedule', extra={'rt_system_id': unicode(self.rt_system_id)}
+            )
+            raise RealtimeProxyError(str(e))
+
+    def _make_url(self, route_point):
+        """
+        The url returns something like a departure on a stop point
+        """
+
+        stop_id = route_point.fetch_stop_id(self.object_id_tag)
+
+        if not stop_id:
+            # one a the id is missing, we'll not find any realtime
+            logging.getLogger(__name__).debug(
+                'missing realtime id for {obj}: stop code={s}'.format(obj=route_point, s=stop_id),
+                extra={'rt_system_id': unicode(self.rt_system_id)},
+            )
+            self.record_internal_failure('missing id')
+            return None
+
+        url = "{base_url}?stop_id={stop_id}".format(base_url=self.service_url, stop_id=stop_id)
+
+        return url
+
+    def _get_dt(self, datetime_str):
+        dt = aniso8601.parse_datetime(datetime_str)
+
+        utc_dt = dt.astimezone(pytz.utc)
+
+        return utc_dt
+
+    def _get_passages(self, route_point, resp):
+        logging.getLogger(__name__).debug(
+            'sytralrt response: {}'.format(resp), extra={'rt_system_id': unicode(self.rt_system_id)}
+        )
+
+        line_code = route_point.fetch_line_id(self.object_id_tag)
+
+        departures = resp.get('departures', [])
+        next_passages = []
+        for next_expected_st in departures:
+            if line_code != next_expected_st['line']:
+                continue
+            dt = self._get_dt(next_expected_st['datetime'])
+            direction = next_expected_st.get('direction_name')
+            is_real_time = next_expected_st.get('type') == 'E'
+            next_passage = RealTimePassage(dt, direction, is_real_time)
+            next_passages.append(next_passage)
+
+        return next_passages
+
+    def _get_next_passage_for_route_point(
+        self, route_point, count=None, from_dt=None, current_dt=None, duration=None
+    ):
+        url = self._make_url(route_point)
+        if not url:
+            return None
+        r = self._call(url)
+
+        if r.status_code != requests.codes.ok:
+            logging.getLogger(__name__).error(
+                'sytralrt RT service unavailable, impossible to query : {}'.format(r.url),
+                extra={'rt_system_id': unicode(self.rt_system_id)},
+            )
+            raise RealtimeProxyError('non 200 response')
+
+        return self._get_passages(route_point, r.json())
+
+    def status(self):
```

Comment:
Not specific to this PR but the status() function is the same for every RT providers. It can be worth moving in the RealtimeProxy class which is the parent of them all

Reply:
True, but it's a bit more work as it also require to move `self.timeout`, `self.breaker` and `self.rt_system_id` to the parent class by adding a `__init__` to the parent class.
I would prefer to do that kind of refactoring in another PR.